src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -458,8 +458,13 @@ public static object ParseFormDataAndDeserialize(Type modelType, string contentT
                 {
                     try
                     {
-                        XmlSerializer serializer = new XmlSerializer(modelType);
-                        serviceModel = serializer.Deserialize(contentStream);
+                        using StreamReader reader = new StreamReader(contentStream, Encoding.UTF8);
+                        string content = reader.ReadToEndAsync().Result;
+                        using (TextReader sr = new StringReader(content))
+                        {
+                            XmlSerializer serializer = new XmlSerializer(modelType);
+                            serviceModel = serializer.Deserialize(sr);
+                        }
```

Comment:
Is this a fix for the not allowed to read sync issue you get in unit tests?
Does this mean we can remove the AllowSynchronousIO = true from startup?
I have solved this in a different way, but only for unit tests by setting AllowSynchronousIO on the TestServer

Reply:
I have not tried to remove this or run other unit tests. I had to change it because it failed since it did not allow to run it synchronous.

Reply:
I've copied this change into the branch I'm currently working on.